asdf/commands/tests/test_extract.py

```diff
@@ -1,11 +1,24 @@
 import os
+import sys
 
 import numpy as np
+import pytest
 from astropy.io.fits import HDUList, ImageHDU
 
 import asdf
 from asdf.commands import extract
-from asdf.fits_embed import AsdfInFits
+from asdf.exceptions import AsdfDeprecationWarning
+
+with pytest.warns(AsdfDeprecationWarning, match="AsdfInFits has been deprecated.*"):
+    # asdf.fits_embed is imported here
+    if "asdf.fits_embed" in sys.modules:
+        del sys.modules["asdf.fits_embed"]
+    # import of asdf.fits_embed is required here to allow for proper removal of
+    # the asdf.fits_embed module (see the above line of code) to trigger the
+    # deprecation warning on import for other tests
```

Comment:
Given this comment, it seems like this module must be run first in order for the other tests to properly function? 

Tests should be independent of order of execution. If this needs to be done in this module then it must be done in every module expecting a similar warning so that they can be run independently.

Reply:
All of the fits_embed imports in the tests have a similar check to allow for independent order of execution.

Reply:
I'm not taking about the check I'm talking about the `import asdf.fts_embed`.

Reply:
I changed the test to use `asdf.fits_embed` instead of `from asdf.fits_embed import AsdfInFits` and removed the comment (as the issues with removing the module caused by the `from` import are no longer relevant).